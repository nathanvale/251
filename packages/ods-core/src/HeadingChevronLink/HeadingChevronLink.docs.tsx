import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import {
  HeadingChevronLinkProps,
  HeadingChevronLink,
} from "./HeadingChevronLink";

export const docs: ComponentDocs<HeadingChevronLinkProps> = {
  category: "Interaction",
  componentName: "HeadingChevronLink",
  description: `By default, this component renders a native "a" element. If youâ€™d like to customise the implementation of this component (e.g. to use a React Router link) you should use ChevronLinkRenderer instead.`,
  propDescriptions: {
    href: "Specifies the URL of the page the link goes to",
  },
  migrationGuide: false,

  examples: {
    default: {
      Code: () => (
        <HeadingChevronLink href="">Heading chevron</HeadingChevronLink>
      ),
    },
  },
  snippets: [
    {
      label: "Default",
      Code: () => (
        <HeadingChevronLink href="">Heading chevron</HeadingChevronLink>
      ),
    },
  ],
};
