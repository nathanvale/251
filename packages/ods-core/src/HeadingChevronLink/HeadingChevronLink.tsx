import React from "react";
import { LinkBaseProps } from "@origin-digital/ods-types";
import { Heading } from "../Heading/Heading";
import { ChevronLinkRenderer } from "../ChevronLinkRenderer/ChevronLinkRenderer";
import { Box } from "../Box/Box";
import { TrackedLink } from "../TrackedLink/TrackedLink";

export interface HeadingChevronLinkProps extends LinkBaseProps {}
const defaultDataId = "heading-chevron-link";
export const HeadingChevronLink = ({
  "data-id": dataId = defaultDataId,
  children,
  domProps,
  target,
  ...rest
}: HeadingChevronLinkProps) => {
  return (
    <ChevronLinkRenderer variant="primary">
      {({ chevronLinkStyles, IconChevron }) => (
        <TrackedLink
          domProps={{ ...domProps, className: chevronLinkStyles }}
          trackingType={HeadingChevronLink.displayName}
          trackingLabel={children}
          data-id={dataId}
          target={target}
          {...rest}
        >
          <Box paddingRight="xsmall">
            <Heading variant="h4">{children}</Heading>
          </Box>
          <IconChevron />
        </TrackedLink>
      )}
    </ChevronLinkRenderer>
  );
};

HeadingChevronLink.defaultProps = {
  "data-id": defaultDataId,
};

HeadingChevronLink.displayName = "HeadingChevronLink";
