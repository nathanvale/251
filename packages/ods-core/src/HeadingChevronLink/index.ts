export { HeadingChevronLink } from "./HeadingChevronLink";
export type HeadingChevronLinkProps = import("./HeadingChevronLink").HeadingChevronLinkProps;
