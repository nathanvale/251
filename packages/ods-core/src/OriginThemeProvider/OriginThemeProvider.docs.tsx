import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Text } from "../Text";
import { TextLink } from "../TextLink";
import { OriginThemeProviderProps } from "./OriginThemeProvider";

export const docs: ComponentDocs<OriginThemeProviderProps> = {
  category: "Logic",
  componentName: "OriginThemeProvider",
  description:
    "OriginThemeProvider lets you apply a consistent tone to your app. It allows you to customize all design aspects of your project in order to meet the specific needs of our brands. By default the OriginThemeProvider comes with the Origin Energy retail theme.",
  propDescriptions: {
    linkComponent:
      "The custom link component you would like to use for ODS link components.",
  },
  migrationGuide: false,
  examples: {
    default: {
      codeString: `import { OriginThemeProvider } from "@origin-digital/ods-core";
export default ({ children }: { children: React.ReactChild }) => (
  <OriginThemeProvider>{children}</OriginThemeProvider>
);
      `,
    },
    additional: [
      {
        label: "Custom link implementation",
        description: (
          <Text>
            By default ODS link components such as{" "}
            <TextLink href="/components/TextLink">TextLink</TextLink>,{" "}
            <TextLink href="/components/ChevronLink">ChevronLink</TextLink> and{" "}
            <TextLink href="/components/HeadingChevronLink">
              HeadingChevronLink
            </TextLink>{" "}
            all use <code>&lt;a/&gt;</code> tags by default. If youâ€™d like to
            customise the implementation of these components (e.g. to use a
            React Router link) you can pass in your own custom link
            implementation.
          </Text>
        ),
        codeString: `
        import React from 'react';
        import { Link as ReactRouterLink } from 'react-router-dom';
        import { OriginThemeProvider } from "@origin-digital/ods-core";
        import { LinkComponentProps } from "@origin-digital/ods-types";

        // First create the custom link implementation:
        const CustomLink = React.forwardRef<any, LinkComponentProps>((props, ref) => {
          const { href, ...restProps } = props;
          return href[0] === "/" ? (
            // For tracking to be active,  
            // You must wrap with React.forardRef and pass down.
            <ReactRouterLink
            to={href}
            ref={ref}
            {...restProps}
            />
            ) : (
            <a href={href} ref={ref} {...restProps} />
          );
        });

        // Then pass it to OriginThemeProvider:
        export const App = () => (
          <OriginThemeProvider linkComponent={CustomLink}>
            ...
          </OriginThemeProvider>
        );
      `,
      },
    ],
  },
  snippets: [],
};
