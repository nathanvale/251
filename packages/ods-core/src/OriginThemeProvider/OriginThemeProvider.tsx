import React, { createContext, useContext } from "react";
import {
  ThemeProvider as MUIThemeProvider,
  Theme as MuiTheme,
} from "@material-ui/core";

import {
  LinkComponentProps,
  LinkComponentType,
} from "@origin-digital/ods-types";
import { coreMuiTheme } from "@origin-digital/ods-themes";
import { LayoutThemeProvider } from "../_private/components/LayoutThemeProvider/LayoutThemeProvider";

const DefaultLinkComponent = React.forwardRef<
  HTMLAnchorElement,
  LinkComponentProps
>((props, ref) => {
  const { children, ...restProps } = props;
  return (
    <a {...restProps} ref={ref}>
      {children}
    </a>
  );
});
const LinkComponentContext = createContext<LinkComponentType>(
  DefaultLinkComponent
);
export const useLinkComponent = () => useContext(LinkComponentContext);

export interface OriginThemeProviderProps {
  children: React.ReactNode;
  linkComponent?: LinkComponentType;
  muiTheme?: MuiTheme;
}

export const OriginThemeProvider = ({
  children,
  muiTheme = coreMuiTheme,
  linkComponent,
}: OriginThemeProviderProps) => {
  const linkComponentFromContext = useLinkComponent();
  return (
    <MUIThemeProvider theme={muiTheme}>
      <LayoutThemeProvider muiTheme={muiTheme}>
        <LinkComponentContext.Provider
          value={linkComponent || linkComponentFromContext}
        >
          {children}
        </LinkComponentContext.Provider>
      </LayoutThemeProvider>
    </MUIThemeProvider>
  );
};
