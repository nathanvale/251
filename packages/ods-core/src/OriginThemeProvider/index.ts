export { OriginThemeProvider } from "./OriginThemeProvider";
export type OriginThemeProviderProps = import("./OriginThemeProvider").OriginThemeProviderProps;
