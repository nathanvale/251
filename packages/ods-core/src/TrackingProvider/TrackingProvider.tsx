import React, { createContext } from "react";
import { TrackingEventHandler } from "@origin-digital/ods-types";

export interface TrackingProviderProps {
  children: React.ReactNode;
  onTrackingCapture: TrackingEventHandler;
}

export const TrackingContext = createContext<
  | undefined
  | {
      onTrackingCapture: TrackingEventHandler;
    }
>(undefined);

export const TrackingProvider = ({
  children,
  onTrackingCapture,
}: TrackingProviderProps) => {
  return (
    <TrackingContext.Provider value={{ onTrackingCapture }}>
      {children}
    </TrackingContext.Provider>
  );
};

TrackingProvider.displayName = "TrackingProvider";
