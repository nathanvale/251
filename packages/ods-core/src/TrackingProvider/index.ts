export { TrackingProvider } from "./TrackingProvider";
export { useTracking } from "./useTracking";

export type TrackingProviderProps = import("./TrackingProvider").TrackingProviderProps;
