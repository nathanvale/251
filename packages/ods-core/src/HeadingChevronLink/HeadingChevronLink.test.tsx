import React from "react";
import { render, fireEvent } from "@origin-digital/ods-testing-library";
import {
  TrackingEventHandler,
  LinkComponentType,
} from "@origin-digital/ods-types";
import { TrackingProvider } from "../TrackingProvider";
import { OriginThemeProvider } from "../OriginThemeProvider";
import { HeadingChevronLink } from "./HeadingChevronLink";

test("It can render", () => {
  const { container } = render(
    <HeadingChevronLink href="">Chevron link</HeadingChevronLink>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it("should call tracking handler when passed into the TrackingProvider", () => {
  const label = "label";

  const onTrackingCapture: TrackingEventHandler = jest.fn();

  const { getByTestId } = render(
    <TrackingProvider onTrackingCapture={onTrackingCapture}>
      <HeadingChevronLink href="">{label}</HeadingChevronLink>
    </TrackingProvider>
  );
  const node = getByTestId("heading-chevron-link");
  fireEvent.click(node);
  expect(onTrackingCapture).toHaveBeenCalledWith({
    "data-id": "heading-chevron-link",
    type: "HeadingChevronLink",
    label,
    postClickState: undefined,
  });
});

test("it handles a custom link implementation", () => {
  const onClick = jest.fn();

  const CustomLink: LinkComponentType = ({ href, ...restProps }) => (
    // eslint-disable-next-line jsx-a11y/anchor-has-content
    <a href={href} onClick={onClick} {...restProps} />
  );

  const { getByText } = render(
    <OriginThemeProvider linkComponent={CustomLink}>
      <HeadingChevronLink href="https://www.test.com/">
        Click me
      </HeadingChevronLink>
    </OriginThemeProvider>
  );

  const link = getByText("Click me");
  fireEvent.click(link);

  expect(onClick).toBeCalledTimes(1);

  expect(getByText("Click me").closest("a")).toHaveAttribute(
    "href",
    "https://www.test.com/"
  );
});
