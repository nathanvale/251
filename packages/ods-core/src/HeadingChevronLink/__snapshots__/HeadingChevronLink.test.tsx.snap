// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can render 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  padding-right: 8px;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<a
  class="typography-linkReset-1 typography-chevron-link-2"
  data-id="heading-chevron-link"
  href=""
>
  <div
    class="c0"
  >
    <h4
      class="typography-h4-variant-basekick-3 typography-neutralDark-color-4 typography-strong-support-5 typography-medium-weight-6 -h4 c1"
      data-id="heading"
    >
      Chevron link
    </h4>
  </div>
  <div
    class="c2"
    style="margin-left: -8px; margin-top: 0px; margin-bottom: 0px; height: 24px;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-8"
      data-id="IconChevronRight"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
</a>
`;
